pinot-broker/src/main/java/org/apache/pinot/broker/requesthandler/BaseBrokerRequestHandler.java

```diff
@@ -438,6 +439,46 @@ public BrokerResponse handleRequest(JsonNode request, @Nullable RequesterIdentit
     return brokerResponse;
   }
 
+  /**
+   * Check if table is in the format of [database_name].[table_name].
+   *
+   * Only update TableName in QuerySource if there is no existing table in the format of [database_name].[table_name],
+   * but only [table_name].
+   *
+   * @param brokerRequest
+   */
+  private void updateQuerySource(BrokerRequest brokerRequest) {
```

Comment:
Should this be done in PinotQuery2BrokerRequestConverter?

Reply:
I wish I could do that!

Here we don't have any information of existing table names as we are allowing dot in table name.

Reply:
Also I feel that we should introduce the concept that no dot is allowed in table name/column name.
I'm not sure if anyone is using dot in table name, but it's definitely some users are using dot in column name.